miniml/compiler/compile.scm

```diff
@@ -353,6 +353,8 @@
     (LPAREN RPAREN) : (list 'PInt 0)
     (LPAREN pattern RPAREN) : $2
     (LBRACE record_list_pattern option_semicolon RBRACE) : (list 'PRecord (reverse $2))
+    (longident_uident DOT LBRACE record_list_pattern option_semicolon RBRACE) :
+        (list 'POpen $1 (list 'PRecord (reverse $4)))
```

Comment:
could you also add the `M.(...)` syntax on patterns, for consistency/symmetry?

Reply:
Done! However, this required a non-trivial change, by changing `'HPConstr` to hold relevant information on the constructor, instead of getting it from the environment. As a bonus, after the matrix decomposition, environments are no longer needed.

Reply:
That sounds good to me, thanks. Two comments:
- Instead of having `tag arity nums` as separate parameters to `HPConstr`, you could just store the `<constr>` record as a single parameter. Even if you prefer this way (it makes pattern-matching slightly more compact), I think it would be better if the parameter order was the same as for the `<constr>` fields -- currently they are mismatched and this is bound to create mistakes down the road.
- Another approach would have beeen to just store the current environment as an extra parameter of `HPConstr` -- but I'm fine with your approach, which is actually nicer I think as we are explicit about which data we really need.

Reply:
I tried to do the change to just storing the `<constr>`, but it doesn't actually work, since `arity` is the actual arity, and not the one for the constructor (relevant in case of `-1` arity, for tuples). I merged the PR as-is, but I'm happy to change the code if you have a better idea.